### PR TITLE
fix embedding bug

### PR DESCRIPTION
**Pull request by roywei**
- fixed embedding layer bug
- used MXNet Embedding layer instead of K.gather 
- Enabled test cases, and tested on end to end example (imdb_fasttext and imdb_cnn)
- imdb_lstm not working due to dropout not supported in lstm layer, created issue [here](https://github.com/awslabs/keras-apache-mxnet/issues/60)
- tested with [external tutorial use case](https://machinelearningmastery.com/use-word-embedding-layers-deep-learning-keras/)
@sandeep-krishnamurthy @karan6181

### CODE REVIEW

File: keras/layers/embeddings.py [hunk 1]
```diff
@@ -135,7 +135,11 @@ def compute_output_shape(self, input_shape):
     def call(self, inputs):
         if K.dtype(inputs) != 'int32':
             inputs = K.cast(inputs, 'int32')
-        out = K.gather(self.embeddings, inputs)
+        # Use K.embedding(mx.sym.Embedding) in MXNet backend instead of K.gather(mx.sym.take)
```
Comment (sandeep-krishnamurthy) [inline keras/layers/embeddings.py]: Please create a Github issue and link it from here.

File: tests/keras/layers/embeddings_test.py [hunk 2]
```diff
@@ -4,8 +4,6 @@
 import keras.backend as K
 
 
-@pytest.mark.skipif((K.backend() == 'mxnet'),
-                    reason='MXNet backend does not support Embeddings yet.')
 @keras_test
```
Comment (sandeep-krishnamurthy) [inline tests/keras/layers/embeddings_test.py]: There are few RNN tests, that are skipped due to Embedding issue. Can you please enable those tests cases as well?

File: keras/backend/mxnet_backend.py [hunk 3]
```diff
@@ -1168,6 +1168,9 @@ def gather(reference, indices):
     indices = mx.sym.Cast(indices.symbol, dtype=reference.dtype)
     return KerasSymbol(mx.sym.take(reference.symbol, indices))
 
+@keras_mxnet_symbol
+def embedding(data, weight, input_dim, output_dim):
```
Comment (sandeep-krishnamurthy) [inline keras/backend/mxnet_backend.py]: Add a check.
if data is KerasSymbol then data.symbol else data
if weight is KerasSymbol then weight.symbol else weight

File: keras/layers/embeddings.py (hunk 1 above)
Comment (roywei) [reply keras/layers/embeddings.py]: created: [here](https://github.com/awslabs/keras-apache-mxnet/issues/63)

File: tests/keras/layers/embeddings_test.py (hunk 2 above)
Comment (roywei) [reply tests/keras/layers/embeddings_test.py]: enabled more test cases

File: tests/integration_tests/test_temporal_data_tasks.py [hunk 4]
```diff
@@ -221,11 +218,7 @@ def test_masked_temporal():
     assert(np.abs(history.history['loss'][-1] - ground_truth) < 0.06)
 
 
-@pytest.mark.skipif(K.backend() != 'tensorflow', reason='Requires TensorFlow backend')
-@pytest.mark.skipif(K.backend() == 'mxnet',
-                    reason='MXNet backend does not support unroll=False '
-                           'in RNN yet.')
-@keras_test
+@pytest.mark.skipif(K.backend() != 'tensorflow' and K.backend() != 'mxnet', reason='Requires TensorFlow or MXNet backend')
```
Comment (sandeep-krishnamurthy) [inline tests/integration_tests/test_temporal_data_tasks.py]: @keras_test is deleted? Please add.
Line length.
